lib/validators/resolveNestedDefinitions.ts

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import {
+  DefinitionsObject,
+  SwaggerObject,
+  ParameterObject,
+  SchemaObject,
+  ResponseObject,
+  PathItemObject,
+  OperationObject
+} from "yasway"
+import { PropertySetTransformation, propertySetMap } from "../util/propertySet"
+import { stringMapMap, stringMapForEach } from "../util/stringMap"
+import { objectPathLast } from "../util/objectPath"
+import { arrayMap } from "../util/array"
+import { Tracked, tracked } from "../util/tracked";
+import * as uuid from "uuid"
+
+export function resolveNestedDefinitions(spec: SwaggerObject): SwaggerObject {
+
+  const newDefinitions: DefinitionsObject = {}
+
+  // a function to resolve nested schema objects
+  function resolveNestedSchemaObject(schemaObjectTracked: Tracked<SchemaObject>) {
```

Comment:
>function res [](start = 2, length = 12)

arrow function for this one ?

Reply:
actually why nested instead of outside and  non exported ?

---
In reply to: [199270718](https://github.com/Azure/oav/pull/268#discussion_r199270718) [](ancestors = 199270718)

Reply:
same for the other ones

---
In reply to: [199270922](https://github.com/Azure/oav/pull/268#discussion_r199270922) [](ancestors = 199270922,199270718)